docs/mdx-plugins/remark-link-rewrite.js

```diff
@@ -0,0 +1,100 @@
+const path = require('path');
+const visit = require('unist-util-visit');
+const { URL } = require('url');
+
+const DEFAULT_OPTIONS = {
+  extension: 'md',
+  pageDir: 'pages',
+  trailingSlash: true,
+};
+
+/**
+ * This rewrites relative MDX links to something Next would expect.
+ *   - Replaces the `.md` extension with a trailing slash.
+ *     e.g. from: /push-notifications/overview.md, to: ./using-fcm.md -> ../using-fcm/
+ *   - Rewrites to absolute path from the page directory (backwards compatibility)
+ *     e.g. from: /push-notifications/overview.md, to ../using-fcm/ -> /push-notifications/using-fcm/
+ *
+ * @param {object} options
+ * @param {string} [options.extension="md"]
+ * @param {string} [options.pageDir="pages"]
+ * @param {boolean} [options.trailingSlash=true]
+ * @returns {function} remark plugin
+ */
+module.exports = function remarkLinkRewrite(options) {
+  const settings = Object.assign({}, DEFAULT_OPTIONS, options);
+
+  return (tree, file) => {
+    // get the relative path, from pages dir to the current file
+    const filePrefix = getFilePrefix(settings, file);
```

Comment:
`filePrefix` could be `undefined` here, right? would it be better to return an empty string in the case that there is no file prefix (i.e. if (fileUrl === ignored))

Reply:
The rewriter now quits early when the file locations aren't defined. 😄